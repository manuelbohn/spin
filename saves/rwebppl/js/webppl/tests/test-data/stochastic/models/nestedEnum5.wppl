var model = function() {
  var x = uniform(0, 1);

  var firstMarginal = Infer({model() {
    var y = flip(x);
    factor(!y ? 0 : -3);
    return y;
  }});

  var secondMarginal = Infer({model() {
    var z = sample(firstMarginal);
    factor(z ? 1 : -1);
    return z;
  }});

  var w = sample(secondMarginal);
  return x * w;
};
