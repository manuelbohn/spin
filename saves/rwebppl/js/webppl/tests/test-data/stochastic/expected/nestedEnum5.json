{
  "mean": 0.245,
  "std": 0.365
}
